File: c#/Clope/Transaction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Clope
{
    internal struct Transaction
    {
        //private List<int> items = new List<int>();    //список id объектов
        private readonly int[] _elementKeys; //список id объектов
        public int ElementKeyCount { get; private set; }  //количество объектов


        public Transaction(int capacity)
        {
            _elementKeys = new int[capacity];
            ElementKeyCount = 0;
        }

        public int this[int index] => _elementKeys[index];

        public void AddElementKey(int idObject) => _elementKeys[ElementKeyCount++] = idObject;        //добавление идентификатора объекта
    }
}

File: Typescript/src/clope/Transaction.ts
import { UID } from "../common/Typings";
export default class Transaction {
    private elementUIDCount: number;

    private readonly elementUIDs: UID[];

    constructor(capacity: number) {
        this.elementUIDs = new Array<UID>(capacity);
        this.elementUIDCount = 0;
    }

    get size(): number {
        return this.elementUIDCount;
    }

    public GetElementUID(num: number): UID {
        return this.elementUIDs[num];
    }

    public AddElementKey(uid: UID): void {
        this.elementUIDs[this.elementUIDCount++] = uid;
    }
}
